crates/starknet-server/tests/test_advancing_time.rs

```diff
@@ -72,22 +67,177 @@ mod advancing_time_tests {
             .await
             .unwrap();
 
-        let new_contract_address = get_udc_deployed_address(
+        get_udc_deployed_address(
             FieldElement::ZERO,
             declaration_result.class_hash,
             &starknet_rs_core::utils::UdcUniqueness::NotUnique,
             &[],
+        )
+    }
+
+    pub async fn get_current_timestamp(
+        devnet: &BackgroundDevnet,
+        contract_address: FieldElement,
+    ) -> FieldElement {
+        let call_current_timestamp = FunctionCall {
+            contract_address,
+            entry_point_selector: get_selector_from_name("get_timestamp").unwrap(),
+            calldata: vec![],
+        };
+        devnet
+            .json_rpc_client
+            .call(call_current_timestamp.clone(), BlockId::Tag(BlockTag::Latest))
+            .await
+            .unwrap()[0]
+    }
+
+    #[tokio::test]
+    async fn timestamp_syscall_set_in_past() {
+        let devnet: BackgroundDevnet =
+            BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let new_contract_address = setup_timestamp_contract(&devnet).await;
+
+        // set time in past
+        let past_time = 1;
+        let set_time_body = Body::from(json!({ "time": past_time }).to_string());
+        let resp_set_time = devnet.post_json("/set_time".into(), set_time_body).await.unwrap();
+        let resp_body_set_time = get_json_body(resp_set_time).await;
+        assert_eq!(resp_body_set_time["block_timestamp"], past_time);
+
+        // mine block
+        devnet.mint(DUMMY_ADDRESS, DUMMY_AMOUNT).await;
```

Comment:
We're really relying a lot on this minting. I suggest that for this test file we create a function called `mine_new_block`, which will still mint, but at least we will have it abstracted, and also there will be no need to write these `// mine block` comments as it will be self-explanatory.

Reply:
fixed in 98d123dbee60863f2483f5fa658d9bc570c5e404